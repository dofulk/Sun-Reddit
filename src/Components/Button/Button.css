.button {
  background-color: rgb(1, 112, 112);
  padding: 10px;
  border-radius: 2px;
  display: inline-block;
  box-shadow: 2px 2px;
  color: white;
  font-size: 26px;
}

.button:hover {
  background-color: rgb(1, 92, 92);
  cursor: pointer;
}