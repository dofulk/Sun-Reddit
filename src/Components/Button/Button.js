import React from 'react';
import './Button.css';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles ({
  root: {
    borderRadius: 2,
    color: 'white',
    backgroundColor: '#005C5C',
    boxShadow: '4px 4px',
    fontSize: '20px',
    '&:hover': {
      backgroundColor: '#006161',
      boxShadow: '6px 6px',
    }
  },
});

const MyButton = (props) => {
  const classes = useStyles();
  return (
    <Button variant="contained" className={classes.root} onClick={props.handleClick} disableElevation>
      {props.label}
    </Button>
  );
} 

export default MyButton;