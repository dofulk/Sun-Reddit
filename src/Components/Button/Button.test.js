import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';


describe('<Button />', () => {
  it('renders correctly with no props', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with all props', () => {
    const tree = renderer.create(<Button label="Press Me" handleClick={() => console.log('clicked')} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should call function on button click', () => {
    const mockHandleClick = jest.fn();
    const tree = renderer.create(
        <Button handleClick={mockHandleClick} />
    )

    renderer.act(() => {
      tree.root.findByType('button').props.onClick();
    })
    expect(mockHandleClick).toHaveBeenCalledTimes(1);
  })

})