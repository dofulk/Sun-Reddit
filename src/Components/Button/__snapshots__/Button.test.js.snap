// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-1 MuiButton-disableElevation"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Press Me
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`renders correctly with no props 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-1 MuiButton-disableElevation"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
</button>
`;
